pkgs/by-name/up/upscayl-ncnn/package.nix

```diff
@@ -0,0 +1,88 @@
+{
+  cmake,
+  fetchFromGitHub,
+  fetchzip,
+  glslang,
+  lib,
+  libwebp,
+  ncnn,
+  stdenv,
+  vulkan-headers,
+  vulkan-loader,
+}:
+# upscayl-ncnn is a fork of /pkgs/by-name/re/realesrgan-ncnn-vulkan, so the nix package is basically the same.
+stdenv.mkDerivation (finalAttrs: {
+  version = "20240601-103425";
+  pname = "upscayl-ncnn";
+
+  src = fetchFromGitHub {
+    rev = finalAttrs.version;
+    hash = "sha256-rGnjL+sU5x3VXHnvuYXVdxGmHdj9eBkIZK3CwL89lN0=";
+    owner = "upscayl";
+    repo = "upscayl-ncnn";
+    fetchSubmodules = true;
+  };
+
+  models = fetchzip {
+    # Choose the newst release from https://github.com/xinntao/Real-ESRGAN/releases to update
+    url = "https://github.com/xinntao/Real-ESRGAN/releases/download/v0.2.5.0/realesrgan-ncnn-vulkan-20220424-ubuntu.zip";
+    stripRoot = false;
+    sha256 = "sha256-1YiPzv1eGnHrazJFRvl37+C1F2xnoEbN0UQYkxLT+JQ=";
+  };
+
+  patches = [
+    ./cmakelists.patch
+    ./models_path.patch
+  ];
+
+  sourceRoot = "${finalAttrs.src.name}/src";
+
+  postPatch = ''
+    substituteInPlace main.cpp --replace REPLACE_MODELS $out/share/models
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    glslang
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeBool "USE_SYSTEM_NCNN" true)
+    (lib.cmakeBool "USE_SYSTEM_WEBP" true)
+
+    "-DGLSLANG_TARGET_DIR=${glslang}/lib/cmake"
+  ];
+
+  buildInputs = [
+    vulkan-loader
+    libwebp
+    ncnn
+    vulkan-headers
+    glslang
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/share
+
+    cp upscayl-bin $out/bin/
+    cp -r ${finalAttrs.models}/models $out/share
+
+    runHook postInstall
+  '';
+
+  postFixup = lib.optionalString stdenv.hostPlatform.isLinux ''
+    patchelf $out/bin/upscayl-bin --add-needed libvulkan.so
+  '';
```

Comment:
I'm not a big fan of this as we've just finished compiling upscayl-bin. Could you try something like

```nix
{
  env.NIX_LDFLAGS = lib.optionalString stdenv.hostPlatform.isLinux "--push-state --as-needed -lvulkan --pop-state";
}
```
or
```nix
{
  env.NIX_LDFLAGS = lib.optionalString stdenv.hostPlatform.isLinux "-rpath ${lib.makeLibrarypath [ vulkan-loader ]}";
}
```

Both of these will do basically the same thing, but just when compiling instead of after

Reply:
Fair point, might be worth applying similar refactors to `realesrgan-ncnn-vulkan` to keep them in sync then. Will probably open a PR on that after this passes review.